src/libraries/System.Private.CoreLib/src/System/Math.DivModInt.cs

```diff
@@ -0,0 +1,224 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+
+#if NATIVEAOT
+using System.Runtime.InteropServices;
+#endif
+
+namespace System
+{
+    /// <summary>
+    /// Provides constants and static methods for trigonometric, logarithmic, and other common mathematical functions.
+    /// </summary>
+    public static partial class Math
+    {
+        [StackTraceHidden]
+        internal static int DivInt32(int dividend, int divisor)
+        {
+            if ((uint)(divisor + 1) <= 1) // Unsigned test for divisor in [-1 .. 0]
+            {
+                if (divisor == 0)
+                {
+                    ThrowHelper.ThrowDivideByZeroException();
+                }
+                else if (divisor == -1)
+                {
+                    if (dividend == int.MinValue)
+                    {
+                        ThrowHelper.ThrowOverflowException();
+                    }
+                    return -dividend;
+                }
+            }
+
+            return DivInt32Internal(dividend, divisor);
+        }
+
+        [StackTraceHidden]
+        internal static uint DivUInt32(uint dividend, uint divisor)
+        {
+            if (divisor == 0)
+            {
+                ThrowHelper.ThrowDivideByZeroException();
+            }
+
+            return DivUInt32Internal(dividend, divisor);
+        }
+
+        [StackTraceHidden]
+        [MethodImpl(MethodImplOptions.AggressiveOptimization)]
+        internal static long DivInt64(long dividend, long divisor)
+        {
+            if ((uint)(int)((ulong)divisor >> 32) == (uint)(int)(((ulong)(int)divisor) >> 32))
```

Comment:
```suggestion
            if ((int)((ulong)divisor >> 32) == (int)(((ulong)(int)divisor) >> 32))
```
Do we need the double `(uint)(int)` cast here? I would expect that one is enough (either `(int)` or `(uint)` should be fine).

Reply:
Yup this cast can be omitted (not sure why analyzer wasn't highlighting this one as it did for other two redundant casts).